chipsec/cfg/pch_3xx.xml

```diff
@@ -71,13 +71,20 @@ XML configuration file for the 300 series PCH
     </register>
 
     <!-- Power Management Controller -->
-    <register name="ABASE" type="pcicfg" bus="0" dev="0x1f" fun="2" offset="0x40" size="4" desc="ACPI Base Address">
-      <field name="STYPE" bit="0" size="1" desc="Space Type (always 1 - I/O space)"/>
-      <field name="BA"    bit="8" size="8" desc="Base Address"/>
+     <register name="ABASE" type="mm_msgbus" port="0x88" offset="0x27B4" size="4" desc="ACPI Base Address">
+      <field name="MASK" bit="18" size="6" desc="MASK for lower 6 bits"/>
+      <field name="BA"    bit="2" size="14" desc="Base Address"/>
+    </register>
+    <register name="PMBASE" type="mm_msgbus" port="0x88" offset="0x27AC" size="4" desc="PM Base Address">
+      <field name="BA" bit="0" size="16" desc="Base Address" />
+      <field name="BAMRL" bit="16" size="16" desc="Memory Range Limit"/>
     </register>
     <register name="PWRMBASE" type="pcicfg" bus="0" dev="0x1f" fun="2" offset="0x10" size="8" desc="PM Base Address">
       <field name="STYPE" bit="0"  size="1"  desc="Space Type (always 0 - memory space)"/>
-      <field name="BA"    bit="13" size="51" desc="Base Address"/>
+      <field name="BA"    bit="12" size="52" desc="Base Address"/>

```

Comment:
I think this is wrong. The definition was fine as it was for 3xx, it's 3xx-OP that has `BA` starting at bit 12. I'd keep this definition as it was, and add the new one to 3xx-op.xml.

Reply:
@ErikBjorge, do you have an opinion on whether we should make a whole new configuration for 3xxOP if the BA for PWRMBASE for 3xx starts at bit 13 (bit 12 will be 0) and the BA for 3xxOP for PWRMBASE starts at bit 12

Reply:
Keep in mind there's also the issue of `PWRMBASE` having bit 31 marked as reserved. I think that'll be a nuisance when trying to read both `PWRMBASE` and `PWRMBASE_HIGH` as one large register.

Reply:
In general if you make the base address field the larger of the two sizes it will work fine and you do not need to create a new configuration file.  This has to do with how BARs are implemented.  Low bits that are not valid will be hard wired to zero to indicate required alignment and size.

With that in mind we should be able to set the start bit to 12 and have it work for both configurations.

Reply:
@ErikBjorge I assume that if we set the start bit to 12 and assume that it's hardwired to 0, we'll still have to modify all the offsets for 3xx to make up for it?

Reply:
It should be handled in the change of the field value seen in the change here since we do not overlap another defined field in the register.

Also, when reading the BA field to get the MMIO base the value is masked and not shifted.  Take a look at read_register_field and the preserve_field_position parameter.  When we get the BA field in the MMIO HAL we set this value to True.

I hope I answered your question.  If not, please clarify the offsets you think need to be updated.

Reply:
Right, I wasn't fully aware of how BARs work. I was under the impression that only the base address field mattered, and we didn't preserve the field position. That makes more sense now.

In any case, bit 31 always being set to reserved/zero is still concerning me. In fact, the whole definition for that register seems to be broken in the 300 On-Package datasheet (at least in DN 337868-002). My other concern is that the `TYPE` field of this BAR is hardwired to zero, meaning that it's 32 bits wide, yet we have `PWRMBASE_HIGH` right after it, making the base address 64 bits wide.

Reply:
Why should bit 31 being forced to 0 concern you?  The OS or any other use of the BAR would just read the value and mask out the lower bits (type and similar).  I have a feeling the reserved high bit is just to prevent overlap with other fixed MMIO registers near 4GB.

The type field should be checked by the PCI enumeration code during resource assignment and the BAR set based on that value (in this case below 4GB).  The fact that we have a potential 64bit address is not a problem.  In the future a version of the chip may support placing the region above 4GB.  The 64bit register definition just supports this flexibility.